# 가설 증명 / TEST

# 카이제곱 검정 
: 두 범주형(category) 변수가 서로 상관이 있는지 판단하는 통계적 검정 방법. <br>
예) 학력, 성별, 직업의 만족도(매우나쁨 (1)~ 매우좋음(5)) 

### 아동별 장난감 보유 현황
- 귀무가설 (당연한 것, 분석할 가치가 없는것) : child 1과 child2의 장난감 보유수에 대한 만족도는 차이가 없다.
- 대립가설 (분석할 것) : child1과 child2의 장난감 보유수에 대한 만독도는 차이가 있다.

In [6]:
# child 별 장난감 보유 현황 데이터
child1 <- c(5,11,1)
child2 <- c(4,7,3)

#분석을 위해 데이터 합치기
Toy <- cbind(child1,child2)
rownames(Toy) <- c("car","truck","doll")
Toy

child1 child2
car    5     4     
truck 11     7     
doll   1     3

In [8]:
# 카이제곱 검정 실시 
chisq.test(Toy)
# 해석 p-value 가 중요하다
cor(child1, child2) # 98퍼 : child1,2가 쌍둥이처럼 비슷한 데이터다. 분석할 이유가 없음

Warning message in chisq.test(Toy):
“Chi-squared approximation may be incorrect”


	Pearson's Chi-squared test

data:  Toy
X-squared = 1.7258, df = 2, p-value = 0.4219


[1] 0.9862414

### 결론
p-value = 0.4219 > 0.05(95% 신뢰구간) -> 귀무가설을 기각할 수 없음 , 대립가설을 체택할 수 없음 <br>
= 데이터의 40%가 사용할 수 없다 : 못 쓰는 데이터다 
> child1과 child2의 데이터는 통계적으로 큰 차이가 없다.

p-value는 0에 가까울수록 좋다. 미리 정해진 유의수준(일반적으로 0.05) 보다 작으면 대립가설을 체택하고 유의수준보다 크면 귀무가설을 체택한다.

### 메세지 : 카이제곱의 근사값이 정확하지 않을 수도 있다.
- 데이터량이 적은 경우 발생

# 피셔검정
데이터 갯수가 적을때 사용

In [10]:
fisher.test(Toy)

# p-value = 0.5165 > 0.05 --> 귀무가설


	Fisher's Exact Test for Count Data

data:  Toy
p-value = 0.5165
alternative hypothesis: two.sided


---
# t - 검정
소비자원 <br>
예) 광고 '배터리 타임 1000시간(평균값)' , 소비자 '배터리 타임 ? '
퀄리티를 일정하게 뽑는 양산기술이 있는지 판단하는데 사용

In [ ]:
# 귀무가설 : 건전지 수명은 1000시간이다.
# 대립가설 : 건전지의 수명은 1000시간이 아니다.

In [14]:
bat <- c(980, 1008, 968, 1032, 1012, 1002, 996, 1017)

# 데이터의 분포가 정규분포인지 확인한다 (진짜 확실한 데이터인지 확인한다) ---> Shapiro-Wilk 검정을 실시
# 귀무가설 : 자료가 정규분포를 따른다
# 대립가설 : 자료가 정규분포를 따르지 않는다

shapiro.test(bat)
# p-value = 0.9758 > 0.05 ---> 귀무가설

# t - 검정 실시
t.test(bat, mu = 1000, alternative = "two.sided")

## t.test 옵션
# mu : 비교하는 대상의 평균
# alternative - two.sided : 데이터가 평균에 따른다
# alternative - greater : 데이터가 평균보다 크다
# alternative - less : 데이터가 평균보다 작다

# p-value = 0.8032 > 0.05 ---> 귀무가설: 건전지의 수명은 1000시간이다.
# 현재 데이터의 평균 1001.875
# 95% 신뢰구간 984.7508 ~ 1018.9992



	Shapiro-Wilk normality test

data:  bat
W = 0.97706, p-value = 0.9469



	One Sample t-test

data:  bat
t = 0.25891, df = 7, p-value = 0.8032
alternative hypothesis: true mean is not equal to 1000
95 percent confidence interval:
  984.7508 1018.9992
sample estimates:
mean of x 
 1001.875 


> 5/4 끗